### PR TITLE
Audio transcription demo

### PR DESCRIPTION
**Pull request by aaron-siegel**
A tutorial notebook demonstrating how to use Pixeltable for audio extraction and transcription.

TODOs (not in this PR):
- Consolidate DocumentSplitter and StringSplitter
- AudioSplitter iterator

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/203)

File: pixeltable/functions/whisper.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    import whisper
+
+    key = (model_id, device)
+    if key not in _model_cache:
+        model = whisper.load_model(model_id, device)
+        _model_cache[key] = model
+    return _model_cache[key]
+
+
+_model_cache = {}
```
Comment (mkornacker) [inline pixeltable/functions/whisper.py]: type hint

File: pixeltable/functions/whisper.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+if TYPE_CHECKING:
+    from whisper import Whisper
+
+
+@pxt.udf(param_types=[pxt.AudioType(), pxt.StringType(), pxt.JsonType(nullable=True), pxt.FloatType(nullable=True),
+                      pxt.FloatType(nullable=True), pxt.FloatType(nullable=True),
+                      pxt.BoolType(), pxt.StringType(nullable=True), pxt.BoolType(), pxt.StringType(),
+                      pxt.StringType(), pxt.StringType(), pxt.FloatType(nullable=True), pxt.JsonType(nullable=True)])
+def transcribe(
```
Comment (mkornacker) [inline pixeltable/functions/whisper.py:13]: Let's move this into functions/openai.py and call it whisper()

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/203)* status: 0 of 14 files reviewed, 2 unresolved discussions (waiting on @mkornacker)

___
*[`pixeltable/functions/whisper.py` line 13 at r3](https://reviewable.io/reviews/pixeltable/pixeltable/203#gh-1653377944:-O0FxrG4AepLX7JC6E1u:b-53p4dn) ([raw file](https://github.com/pixeltable/pixeltable/blob/6b2be6aef914f45c75cd4460d734f567783b529b/pixeltable/functions/whisper.py#L13)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Let's move this into functions/openai.py and call it whisper()
</blockquote></details>

I don't agree with this, for a bunch of reasons:
- everything in `functions.openai` is an API endpoint; whisper is something fundamentally different (an apache open source library that happens to be maintained by OpenAI, vs. remote endpoints for OpenAI the cloud service)
- it's important to preserve the convention that Pixeltable function names mirror the underlying libraries, as much as we can; otherwise it'll be really hard to stay consistent
___
*[`pixeltable/functions/whisper.py` line 64 at r3](https://reviewable.io/reviews/pixeltable/pixeltable/203#gh-1653375773:-O0FxXkC1npAgdK85Ezu:bb6un1j) ([raw file](https://github.com/pixeltable/pixeltable/blob/6b2be6aef914f45c75cd4460d734f567783b529b/pixeltable/functions/whisper.py#L64)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

type hint
</blockquote></details>

Done

**Review by mkornacker — APPROVED** [review]

**Pull request merged**